src/ConnectedMode/ServerSentEvents/IServerSentEventSource.cs

```diff
@@ -26,21 +26,22 @@ namespace SonarLint.VisualStudio.ConnectedMode.ServerSentEvents
     /// <summary>
     /// Source for the server sent events about certain server side changes to the SQ/SC project
     /// </summary>
+    /// <remarks>This interface is not intended to be thread safe</remarks>
     /// <typeparam name="T">Server sent event type inherited from <see cref="IServerEvent"/></typeparam>
-    internal interface IServerSentEventSource<T> where T : IServerEvent
+    internal interface IServerSentEventSource<T> where T : class, IServerEvent
```

Comment:
is it important that it's class?

Reply:
Yes, otherwise the event source can't return null

Reply:
https://media1.giphy.com/media/3og0IPBdowkBBNtKTK/giphy.gif